Plugin/Block/Product/Image.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Fisheye Media Ltd. All rights reserved.
+ * See LICENCE.txt for licence details.
+ */
+namespace Fisheye\Lazyload\Plugin\Block\Product;
+
+
+class Image
+{
+
+    /**
+     * Plugin for lazyloading images - Set template to Fisheye_Lazyload
+     *
+     * @param \Magento\Catalog\Block\Product\Image $subject
+     * @param $template
+     * @return array
+     */
+    public function beforeSetTemplate(\Magento\Catalog\Block\Product\Image $subject, $template) {
+        $template = str_replace('Magento_Catalog', 'Fisheye_Lazyload', $template);
+        return [$template];
```

Comment:
The sole parameter for `setTemplate()` in `Magento\Framework\View\Element\Template` is a string, so we should also send a string, not an array

Reply:
Won't work sending back a string, needs to be an array for some reason. Followed same process as outline here: http://devdocs.magento.com/guides/v2.1/extension-dev-guide/plugins.html#before-methods

Reply:
Fair enough, that'll be me not spending enough time with `before` plugins 👍